cmd/maestro/server/grpc_server.go

```diff
@@ -150,8 +176,19 @@ func (svr *GRPCServer) Publish(ctx context.Context, pubReq *pbv1.PublishRequest)
 			return nil, fmt.Errorf("failed to update resource: %v", err)
 		}
 	case common.DeleteRequestAction:
-		err := svr.resourceService.MarkAsDeleting(ctx, res.ID)
+		res, err = decode(eventType.CloudEventsDataType, evt)
 		if err != nil {
+			return nil, fmt.Errorf("failed to decode cloudevent: %v", err)
+		}
+		var deletionTimestamp time.Time
+		evtExtensions := evt.Context.GetExtensions()
+		if _, ok := evtExtensions[codec.ExtensionWorkMeta]; ok {
+			if _, exists := evtExtensions[types.ExtensionDeletionTimestamp]; exists {
+				deletionTimestamp = time.Now()
+			}
```

Comment:
@skeeey is `evtExtensions[types.ExtensionDeletionTimestamp]` set by the source client always when deleting a manifest work?

Reply:
yes, it is.

Reply:
Thanks for confirming.